modules/geo-layers/src/quadkey-layer/quadkey-layer.js

```diff
@@ -0,0 +1,87 @@
+import {CompositeLayer} from '@deck.gl/core';
+import {PolygonLayer} from '@deck.gl/layers';
+import {getQuadkeyPolygon} from './quadkey-utils';
+
+const defaultProps = {
+  ...PolygonLayer.defaultProps,
+  getQuadkey: {type: 'accessor', value: d => d.quadkey}
+};
+
+export default class QuadkeyLayer extends CompositeLayer {
+  renderLayers() {
+    // Layer prop
```

Comment:
A lot of this code is duplicated with `S2Layer`. It may be worth considering introducing a common subclass

Reply:
Agreed

Reply:
As the shared class wouldn't be ever exposed I instead used a utility method as it felt more consistent with what with do with `GeoJsonLayer`